lib/pact/consumer_contract/request_decorator.rb

```diff
@@ -20,25 +20,35 @@ def as_json options = {}
       hash[:query]   = query   if request.specified?(:query)
       hash[:headers] = headers if request.specified?(:headers)
       hash[:body]    = body    if request.specified?(:body)
-      hash
+      include_matching_rules? ? with_matching_rules(hash) : hash
     end
 
     private
 
     attr_reader :request
 
     def path
```

Comment:
Pact::Reification.from_term(request.query) for instance takes the term that defines the matching rule and wipes it out leaving you with the the example value passed with the term. When I get the time again at work I will write a test to exercise this functionality. It is not in my nature to leave functionality untested but this was my first ruby experience and I was a little short on time. I'll notify you when I get back to this.